## class desc
#@icon("")
#class_name XXX
extends RichTextLabel


#region SIGNALS

#endregion


#region ON READY (for direct children only)

#endregion


#region EXPORTS
@export_group("Component Links")
@export var sim: Sim = null
#
# @export_group("Details")
#endregion


#region VARS

#endregion


#region FUNCS
func _process(delta: float) -> void:
	text = ""
	for person in sim.people:
		var alive_text: String = ""
		if person.is_alive:
			alive_text = "ğŸ§"
		else:
			alive_text = "ğŸ’€"

		text += str(
			person.job, " | ",
			alive_text, " | ",
			"â¤ï¸ ", person.health, " | ",
			"ğŸª™ ", person.stockpile["money"], " | ",
			"ğŸ¥ª ", person.stockpile["grain"], " | ",
			"ğŸ’§ ", person.stockpile["water"], " | ",
			"ğŸªµ ", person.stockpile["wood"], " | ",
			"\n"
		) #"ğŸªµâ¤ï¸ğŸ’§ğŸ¥ª"








#endregion
